File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution(object):
    def evalRPN(self, tokens):
        """
        :type tokens: List[str]
        :rtype: int
        """
        stack = []
        for token in tokens:
            if token.isdigit() or (token[0] == '-' and len(token) > 1):
                stack.append(int(token))
            else:
                if len(stack) < 2:
                    raise ValueError("Insufficient operands for operator")
                val1 = stack.pop()
                val2 = stack.pop()

                if token == "+":
                    stack.append(val2 + val1)
                elif token == "-":
                    stack.append(val2 - val1)
                elif token == "*":
                    stack.append(val2 * val1)
                elif token == "/":
                    stack.append(int(float(val2) / val1))  # Integer division
        return stack[0]


                
                
            

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cs
public class Solution {
    public int EvalRPN(string[] tokens) {
        Stack<int> st = new Stack<int>();

        foreach (string c in tokens){
            if (c == "+"){
                int first = st.Pop();
                int second = st.Pop();
                int addition = first + second;
                st.Push(addition);
            } else if (c == "-"){
                int first = st.Pop();
                int second = st.Pop();
                int sub = second - first;
                st.Push(sub);
            } else if (c == "*"){
                int first = st.Pop();
                int second = st.Pop();
                int mult = first * second;
                st.Push(mult);
            } else if (c == "/"){
                int first = st.Pop();
                int second = st.Pop();
                int division = 0;
                if (second == 0){
                    division = 0;
                } else {
                    division = second / first;
                }
                st.Push(division);
            } else {
                st.Push(int.Parse(c));
            }
        }
        return st.Peek();
        
    }
}